app/src/main/java/com/lastaoutdoor/lasta/data/db/ActivitiesDBRepositoryImpl.kt

```diff
@@ -101,6 +104,20 @@ constructor(context: Context, database: FirebaseFirestore) : ActivitiesDBReposit
         .await()
   }
 
+  override suspend fun updateDifficulty(activityId: String) {
+    val document = activitiesCollection.document(activityId)
+    val activity = document.get().await()
+    val difficulty = activity.getString("difficulty") ?: "EASY"
+    val newDifficulty =
+        when (difficulty) {
+          "EASY" -> "NORMAL"
+          "NORMAL" -> "HARD"
+          "HARD" -> "EASY"

```

Comment:
Maybe you can add this function as a member of the Difficulty class to modularize the code a bit more

Reply:
It's a good idea, however I'm using this only once and it won't be used again so maybe not really needed in this case !